packages/components-native/src/Banner/Banner.tsx

```diff
@@ -17,43 +18,79 @@ export function Banner({
   children,
   icon,
 }: BannerProps): JSX.Element {
+  const bannerIcon = icon || getBannerIcon(type);
+
+  const shouldFlow =
+    Boolean(React.Children.count(children) === 1 && !text && !details) ||
+    Boolean(text && !details && !children);
+
   return (
     <Pressable
-      style={[styles.container, BannerTypeStyles[type].styles]}
+      style={[styles.container]}
       accessibilityRole="alert"
       onPress={action?.onPress}
     >
       <Content childSpacing="small">
         <View style={styles.bannerContent}>
-          {icon && <BannerIcon icon={icon} />}
+          {bannerIcon && <BannerIcon icon={bannerIcon} type={type} />}
           <View style={styles.contentContainer}>
             <View style={styles.childrenContainer}>
-              <BannerChildren>{children}</BannerChildren>
+              <WrappingElement shouldFlow={shouldFlow}>
+                <BannerChildren>{children}</BannerChildren>
+
+                {text && <Text level="text">{text}</Text>}
+
+                {details && <TextList items={details} level="text" />}
+
+                {action && (
+                  <RNText>
+                    {shouldFlow && <Typography> | </Typography>}
```

Comment:
@taylorvnoj is it easy enough to make this typography subdued?

Reply:
Done ✔️  @fagnanm